python/ray/data/_internal/execution/streaming_executor.py

```diff
@@ -295,13 +301,14 @@ def _scheduling_loop_step(self, topology: Topology) -> bool:
         for op_state in topology.values():
             op_state.refresh_progress_bar()
 
-        if not DEBUG_TRACE_SCHEDULING:
-            _debug_dump_topology(topology, log_to_stdout=False)
-
-        update_stats_actor_metrics(
-            [op.metrics for op in self._topology],
-            self._get_metrics_tags(),
-        )
+        if time.time() - self._last_stats_update >= STATS_ACTOR_UPDATE_INTERVAL_SECONDS:
```

Comment:
Yes, I think making the last_stats_update a global variable would be fine here.

Reply:
There are still two methods `update_stats_actor_metrics` and `update_stats_actor_iter_metrics`, so should we have two separate global variables? Since we need to make the iter path threadsafe, and the execution path is coarser, is it possible that the iter path can hog the updates if we have a single variable?